src/lib/multipageloader.cc

```diff
@@ -107,6 +107,20 @@ QNetworkReply * MyNetworkAccessManager::createRequest(Operation op, const QNetwo
 	return QNetworkAccessManager::createRequest(op, r3, outgoingData);
 }
 
+MyNetworkProxyFactory::MyNetworkProxyFactory (QNetworkProxy proxy, QList<QString> bph, ResourceObject & res):
+	defaultProxy(proxy),
+	bypassHosts(bph),
+        resource(res) {}
+
+QList<QNetworkProxy> MyNetworkProxyFactory::queryProxy (const QNetworkProxyQuery & query) {
+	QString host = query.url().host();
+	foreach (const QString & bypassHost, bypassHosts) {
+		if (host == bypassHost) {
```

Comment:
Should ideally use `host.compare(bypassHost, Qt::CaseInsensitive)` as then `google.com` and `GOOGLE.COM` will match -- DNS is normally case-insensitive.